app/src/main/java/com/futurice/freesound/feature/common/ui/adapter/RecyclerViewAdapter.java

```diff
@@ -38,10 +38,10 @@
 /**
  * Implementation of {@link RecyclerView.Adapter} for {@link DisplayableItem}.
  */
-public final class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+public final class RecyclerViewAdapter<T> extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
```

Comment:
IMO when you want to have different items in one adapter you have two approaches:
- Make it type unsafe with the wrapping approach here: the good about this is that you remove the boiler plate of having to create interfaces.
- Create interfaces for all the elements in a given list

Not sure what option is best, but maybe we should discuss about it before deciding